distance-sensor.c

```diff
@@ -57,6 +61,21 @@ uint16_t getTimerTicks() {
 }
 
 float calculateDistance() {
-    // clockTicks / 2(account for twice distance) * speed of sound(m/s) / clockFrequency(500kHz)
-    return timerTicks / 2.0 * 343 / 500000;
+    if (counter == 256) {                           // Sets counter to zero when distance[] overflows]
+        counter = 0;
+        toggle = 0;                                 // Ensures average is divided by 256 with array is full
+    }
+    for (int i = 0; i < 16; i++) {                  // Will update distance[] 16 times before average is returned
+        average -= distance[counter];               // Delete current timerTick from average
+        distance[counter] = timerTicks;             // Update distance[]
+        average += distance[counter];               // Add current timerTick to average
+        counter++;                                  // Increment counter
+    }
+
+    // clockTicks / number of slots filled in distance[] / 2(account for twice distance) * speed of sound(m/s) / clockFrequency(500kHz)
+    if (toggle) {
```

Comment:
Great that you took this into account so we don't have to wait for the buffer to fill up in order to get stable values!